File: m3-h2/src/main/java/ru/yegorpilipenko/otus/spring/m3h2/service/AuthorServiceImpl.java
package ru.yegorpilipenko.otus.spring.m3h2.service;

import lombok.RequiredArgsConstructor;
import org.springframework.security.access.annotation.Secured;
import org.springframework.stereotype.Service;
import ru.yegorpilipenko.otus.spring.m3h2.model.Author;
import ru.yegorpilipenko.otus.spring.m3h2.repository.BookRepository;

import java.util.Collection;

import static ru.yegorpilipenko.otus.spring.m3h2.model.User.Role.ADMIN;
import static ru.yegorpilipenko.otus.spring.m3h2.model.User.Role.USER;

@Service
@Secured({ADMIN, USER})
@RequiredArgsConstructor
public final class AuthorServiceImpl implements AuthorService {

    private final BookRepository bookRepository;

    @Override
    public Collection<Author> getAll() {
        return bookRepository.findAllAuthors();
    }

}

File: m1-h5/src/main/kotlin/ru/yegorpilipenko/otus/spring/m1h5/service/AuthorServiceImpl.kt
package ru.yegorpilipenko.otus.spring.m1h5.service

import org.springframework.stereotype.Service
import ru.yegorpilipenko.otus.spring.m1h5.dao.AuthorDao
import ru.yegorpilipenko.otus.spring.m1h5.model.Author

@Service
class AuthorServiceImpl(val authorDao: AuthorDao): AuthorService {

    override fun findAll(): Collection<Author> = authorDao.findAll()

    override fun findById(id: Long): Author? = authorDao.findById(id)

    override fun save(author: Author): Author = authorDao.save(author)

    override fun removeById(id: Long): Boolean = authorDao.removeById(id)

}
